plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 testEmptyVolumeConfig,
+				dailyVolumeByDateQuery: emptyConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - market ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{additionalMarketIDs: marketIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{additionalMarketIDs: marketIDs},
+				dailyVolumeByDateQuery: marketConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - account ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{optionalAccountIDs: accountIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs},
+				dailyVolumeByDateQuery: accountConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - account and market ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+				dailyVolumeByDateQuery: accountMarketConfigVolumeQuery,
+			},
+		},
+	}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			actual, e := makeFilterVolume(
+				k.configValue,
+				k.exchangeName,
+				k.tradingPair,
+				k.assetDisplayFn,
+				k.baseAsset,
+				k.quoteAsset,
+				k.db,
+				k.config,
+			)
+
+			assert.Equal(t, k.wantError, e)
+			assert.Equal(t, k.wantSubmitFilter, actual)
+		})
+	}
+}
+
+func TestVolumeFilterFn(t *testing.T) {
+	db := connectTestDb()
```

Comment:
we don't need a db here -- we can manually construct a `volumeFilter` by passing in `nil` for the `dailyVolumeByDateQuery` since that's not used. Really what's actually used in the function is the `config` and the base/quote assets. I'm ok with changing the signature of `volumeFilterFn` to take in only those fields and making it a function that is not hanging off the volumeFilter. This will ensure we don't need to construct excessive objects to test it